packages/preview/t4t/0.4.1/t4t-manual.typ

```diff
@@ -0,0 +1,238 @@
+
+#import "@preview/mantys:0.1.4": *
+#import "@preview/tidy:0.3.0"
+
+#import "tools4typst.typ"
+
+#let module-scope = (
+  t4t: tools4typst,
+  test: tools4typst.test,
+  def: tools4typst.def,
+  assert: tools4typst.assert,
+  get: tools4typst.get,
+  math: tools4typst.math,
+)
+
+#let show-tests(display: false, ..tests) = {
+  if display {
+    let code = tests.pos().fold(
+      "",
+      (c, r) => {
+        c += r.text + "\n"
+        c
+      },
+    )
+    scale(x: 75%, y: 75%, origin: left)[
+      *Tests*\
+      #v(-1.25em)
+      #codesnippet(raw(lang: "typc", code))
+    ]
+  }
+}
+
+#let test(scope: (:), ..tests) = {
+  for test in tests.pos() {
+    let msg = if test.text.starts-with("not ") {
+      test.text.slice(4) + " should be false, was true"
+    } else {
+      test.text + " should be true, was false"
+    }
+    assert(
+      eval(test.text, scope: module-scope + scope),
+      message: msg,
+    )
+  }
+  show-tests(..tests)
+}
+
+#let show-module(name) = module-commands(name)[
+  #tidy-module(
+    read(name + ".typ"),
+    name: name,
+    scope: module-scope + (utest: test),
+    // tidy: tidy,
+  )
+]
+
+
+#show: mantys.with(
+  ..toml("typst.toml"),
+
+  subtitle: [Tools For Typst],
+  date: datetime.today(),
+  abstract: [
+    *Tools for Typst* (`t4t` in short) is a utility package for Typst package and template authors. It provides solutions to some recurring tasks in package development.
+
+    The package can be imported or any useful parts of it copied into a project. It is perfectly fine to treat `t4t` as a snippet collection and to pick and choose only some useful functions. For this reason, most functions are implemented without further dependencies.
+
+    Hopefully, this collection will grow over time with *Typst* to provide solutions for common problems.
+  ],
+)
+
+= Usage
+
+== Load from package repository (Typst 0.6.0 and later)
+
+For Typst 0.6.0 and later, the package can be imported from the _preview_ repository:
+
+#codesnippet[```typ
+  #import "@preview/t4t:0.3.1": automaton
```

Comment:
I didn't notice last time, but this version is outdated (there a few other occurrences below as well).